### DISCUSSION TITLE
Quota option

### DISCUSSION THREAD

**Discussion by lan10rd** [trapexit/mergerfs#919] [Ideas]
Is it possible / feasible for a quote option? maybe in mergerfs-tools?  Likewise, reservations an option?  For the life of me I am not finding an easy way to set quotas to folders, like  users/a - 1gb users/b - 500mb, organization/a 2gb.

everybody online is like fstab this with quota or whatever. i use zfs underneath so there is that as a last resort, but meh i would prefer it at the mergerfs level, idc if its costly to performance to check before right or whatever, this would save my ass again (mergerfs already has)

**Comment by trapexit** [comment]
A per directory quota? That's really non-trivial given how mergerfs works. It doesn't have access to information sufficient to make that decision without scanning the all the filesystems at startup, assuming nothing changes out of band, and keeping track of all actions that impact size. It'd be way easier to just allocate volumes at the size you want and mount them in the appropriate spot.

**Reply by lan10rd** [reply]
okie dokie just checking, thanks you da man (or woman)?! i was watching for comment notifications wow can't believe you responded so fast i didn't even notice.  tyty forever

**Reply by trapexit** [reply]
No problem. You can take a look into kernel level quotas but I'm not sure it offers what you're looking for. Off hand I can't recall of anything having per directory quotas. When I worked on a proprietary container orchestration platform years ago we used LVM to create temporary space on demand because it was the most practical way to create "quotas".  I'm not sure if anything has changed significantly in that space.

**Reply by lan10rd** [reply]
so im doing zfs for this where you can create datasets, and because im kinda limited by usb hubs, im creating zfs pools on separate pis, mounting pi a’s zfs pool and b’s pool on each other, than mergerfs into a mergedzpool so that i can have a zfs nas. zfs for redundany plus i really needed a way to not let users go crazy and blow past a limit.  im still assuming i need moveonenospc=false right? otherwise by virtue of no soace left or quota met mergerfs may try to find another spot?

**Reply by lan10rd** [reply]
the idea is zpools/a/users/abc and zpools/b/users/def. where abc could have a 500mb quota and def could have 1gb quota, but the web apps and containers can just mount the mergedzpools folder as if all users were on the same drive.  im sure you get the point hehe
